### ISSUE TITLE
Gameyfin does not recognize most retro game rom files

### ISSUE THREAD

**Issue by CraftyCanine** [grimsi/gameyfin#92]
Gameyfin does not recognize retro game rom files from the vast majority of retro systems. This project is perfect for a retro games presentation tool, and such a use case would align really well with the use case of Plex and Jellyfin, but it's incredibly limited by the super small set of file extensions it recognizes. A small change would greatly expand the scope of the project.

From [this](https://github.com/grimsi/gameyfin/blob/424ff805d40789449b2e955c06e91e0c484e0db3/backend/src/main/resources/config/gameyfin.yml) file:
`file-extensions: iso, zip, rar, 7z, exe`

Compiling a list from the internet such as [this](https://emulation.gametechwiki.com/index.php/List_of_filetypes) one and just adding them in would be an easy fix.

**Comment by CraftyCanine** [comment]
Another option would be to move this out into the user config and let the users specify what file extensions they want the app to recognize.

I think the most appropriate would probably be to bundle the extensions with each game library as an option, so that each library only supports the type of file/files that the roms are typically stored as.

**Comment by grimsi** [comment]
Since Gameyfin is using the Spring Boot framework, you can already overwrite the file-extensions list (see [here](https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config)).

The easiest way is probably by setting an environment variable like this:
GAMEYFIN_FILE_EXTENSIONS=<Comma seperated list of your file extensions>

Another possibility would be to create a file named `application.properties` with the following content and placing it next to the `gameyfin.jar` file:
```
gameyfin.file-extensions=<Comma seperated list of your file extensions>
```

I think adding the list from gametechwiki can be done so you there would be no need to modify the configuration manually.
Also bundling the extensions with the libraries is not really possible, since the library types are fetched from IGDB and IGDB does not (to my knowledge) provide information on file types.

**Comment by CraftyCanine** [comment]
Thank you for the quick reply! I have some familiarity with spring boot, I’ll have to try that. Definitely a perfectly fine workaround at the very least.

However, I do think it would make sense in the configuration alongside the library mappings. I don’t know exactly how things are stored on the backend, but I think it would fit really nicely alongside the library slug in the UI at least.

**Comment by grimsi** [comment]
Fixed in #149 (Release 1.4.0)
